src/components/patch-assignment-form/patch-assignment-form.tsx

```diff
@@ -0,0 +1,333 @@
+    return (
+      <Thead>
+        <Tr>
+          <Th>Patch</Th>
+          <Th>Area</Th>
+          <Th>Assigned Officer</Th>
+          <Th>Contact</Th>
+          <Th />
+        </Tr>
+      </Thead>
+    );
+  };
+
+  const PatchTableBody = ({ tableItems }: { tableItems: Patch[] }): JSX.Element => {
+    let patches = tableItems.filter((patchOrArea) => patchOrArea.patchType === "patch");
+    const areas = tableItems.filter((patchOrArea) => patchOrArea.patchType === "area");
+
+    interface PatchTableItem extends Patch {
+      parentAreaName: string | undefined;
+    }
+
+    let patchTableItems: PatchTableItem[] = [];
+    if (areaOption === "all") {
+      patchTableItems = tableItems as PatchTableItem[];
+    } else {
+      const selectedArea = areas.find((area) => area.name === areaOption);
+
+      patches = patches.filter((patch) => patch.parentId === selectedArea?.id);
+      patchTableItems.push(selectedArea as PatchTableItem);
+      patches.forEach((patch) => {
+        patchTableItems.push(patch as PatchTableItem);
+      });
+    }
+
+    patches.forEach((patch) => {
+      const patchListItem = patch as PatchTableItem;
+      const parentArea = areas.filter((area) => area.id === patch.parentId)[0];
+      patchListItem.parentAreaName = parentArea?.name;
+    });
+
+    patchTableItems = patchTableItems.sort((a, b) => (a.name > b.name ? 1 : -1));
+
+    patchTableItems.sort((a, b) => {
+      return a.patchType !== "area" && b.patchType === "area" ? 1 : -1;
+    });
+
+    /**
+     * Shows reassign btn if not reassigning, shows cancel btn if reassigning this patch, shows assign btn if reassigning another patch
+     * @param patch - A patch or area the button is for
+     * @returns a JSX.Element button
+     */
+    const DisplayedButton = ({ patch }: { patch: Patch }) => {
+      if (!reassigningPatch) {
+        return (
+          <button
+            data-testid="reassign-button"
+            className="govuk-button lbh-button"
+            style={{ marginTop: 0, width: "10em" }}
+            onClick={(e) => {
+              e.preventDefault();
+              setAreaOption("all");
+              setReassigningPatch(patch);
+            }}
+          >
+            Reassign
+          </button>
+        );
+      }
+      const reassigningThisEntity = patch.id === reassigningPatch.id;
+      if (reassigningThisEntity) {
+        return <CancelReassignmentButton />;
+      }
+      const officerBeingAssignedFirstName =
+        reassigningPatch?.responsibleEntities[0]?.name.split(" ")[0];
+      return (
+        <button
+          data-testid="assign-button"
+          className="govuk-button lbh-button"
+          style={{ marginTop: 0, maxHeight: "2.5em" }}
+          onClick={(e) => {
+            e.preventDefault();
+            setSwitchingWithPatch(patch);
+            setDialogActive(true);
+          }}
+        >
+          Assign {officerBeingAssignedFirstName}
+        </button>
+      );
+    };
```

Comment:
Do you think this component name is descriptive enough? I think this would be more readable as two separate components.

Reply:
Fair point, it had a lot going on.
I split it up a bit in 8a23c86755686cfc251123ca89453deff2ce853c